src/netInfo.re

```diff
@@ -0,0 +1,83 @@
+type infoType = string;
+
+type infoEffectiveType = string;
+
+type connectionType =
+  | None
+  | WiFi
+  | Cellular
+  | Unknown
+  | Bluetooth
+  | Ethernet
+  | WiMax;
+
+type effectiveConnectionType =
+  | Net2G
+  | Net3G
+  | Net4G
+  | Unknown;
+
+type info = {
+  .
+  "effectiveType": infoEffectiveType,
+  "_type": infoType
+};
+
+let connectionType = connectionType =>
+  switch connectionType {
+  | "none" => None
+  | "wifi" => WiFi
+  | "cellular" => Cellular
+  | "unknown" => Unknown
+  | "bluetooth" => Bluetooth
+  | "ethernet" => Ethernet
+  | "wimax" => WiMax
+  | connection =>
+    failwith("NetInfo, unexpected `connection` type: " ++ connection)
+  };
+
+let effectiveConnectionType = effectiveConnectionType =>
+  switch effectiveConnectionType {
+  | "2g" => Net2G
+  | "3g" => Net3G
+  | "4g" => Net4G
+  | "unknown" => Unknown
+  | connection =>
+    failwith("NetInfo, unexpected `effectiveConnection` type: " ++ connection)
+  };
+
+[@bs.module "react-native"] [@bs.scope "NetInfo"]
+external addEventListener :
+  ([@bs.as "connectionChange"] _, info => unit) => unit =
+  "";
+
+[@bs.module "react-native"] [@bs.scope "NetInfo"]
+external removeEventListener :
+  ([@bs.as "connectionChange"] _, info => unit) => unit =
+  "";
+
+[@bs.module "react-native"] [@bs.scope "NetInfo"]
+external isConnectionExpensive : unit => Js.Promise.t(bool) = "";
+
+[@bs.module "react-native"] [@bs.scope "NetInfo"]
+external getConnectionInfo : unit => Js.Promise.t(info) = "";
+
+module IsConnected = {
+  type t;
+  [@bs.module "react-native"] [@bs.scope "NetInfo"] [@bs.val]
+  external isConnected : t = "";
+  [@bs.send.pipe : t]
+  external _addEventListener :
+    ([@bs.as "connectionChange"] _, bool => unit) => unit =
+    "addEventListener";
+  [@bs.send.pipe : t]
+  external _removeEventListener :
+    ([@bs.as "connectionChange"] _, bool => unit) => unit =
+    "removeEventListener";
+  [@bs.send.pipe : t] external _fetch : unit => Js.Promise.t(bool) = "fetch";
```

Comment:
`fetch` [is deprecated](https://facebook.github.io/react-native/docs/netinfo.html#fetch) or `getConnectionInfo` doesn't exist depending which version you target. Managing that is major PITA with building bindings.

Reply:
hm, according to the source `NetInfo.isConnected.fetch` is not deprecated, take a look here:
https://github.com/facebook/react-native/blob/master/Libraries/Network/NetInfo.js#L242

and `NetInfo.fetch` is deprecated 😉 
https://github.com/facebook/react-native/blob/master/Libraries/Network/NetInfo.js#L182

> Managing that is major PITA with building bindings.

is there any solution to handle this?

Reply:
@mobily this is a good question :). I don't have a good answer for that. Feel free to share ideas.